walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,87 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.LinkedHashSet;
+import java.util.List;
+
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    // Collection<TimeRange> meetingTimes = new HashSet<>();
+    Collection<TimeRange> meetingTimes = new LinkedHashSet<>();
+    int start = TimeRange.START_OF_DAY;
+    int end = TimeRange.END_OF_DAY;
+    int freeTime;
+    List<TimeRange> unavailableTimes = new ArrayList<>();
```

Comment:
are any of these variables used across steps of the for loop? if not, please instantiate them in the loop so that their scopes are only in the step of the for loop where it's used.

Reply:
I use start and end inside and outside of the loop, does that change anything?

Reply:
Do I still instantiate inside the loop too?

Reply:
it depends.
- if step `i` needs the value of variable `x` from step `i-1`, then you should declare `x` outside of the loop.
- if after the loop completes, you need the final value of `x`, then you should also declare it outside of the loop.

it looks like from your code, `start` is set at the end of the iteration, and the value is used in the next one, so you'd need to declare it outside.
it looks like for each iteration, you do `end = time.start();` before `end` is ever used, so you can declare it inside.

Reply:
I see, that makes sense!